server.js

```diff
@@ -0,0 +1,291 @@
+const http = require('http');
+const fs = require('fs');
+const qs = require('querystring');
+const userPasswords = require('./userPasswords.js');
+
+const server = http.createServer((req, resp) => {
+  if (req.method === 'PUT') {
+    const headerObj = req.headers;
+    const hasAuthHeader = headerObj.hasOwnProperty('authorization');
+    if (!hasAuthHeader) {
+      resp.writeHead(401, {
+        'Content-Type': 'application/json',
+        'WWW-Authenticate': `Basic realm="Secure Area"`,
+      });
+      resp.end(`<html><body>Not Authorized</body></html>`);
+    } else if (hasAuthHeader) {
```

Comment:
If you put the return in front of your resp.end above, then you don't need the else if here and can move all the code in the body of your else if back one indent.